diff --git a/tests/snapshot/bytecode/class/methods.exp b/tests/snapshot/bytecode/class/methods.exp
index 1341994..b16d324 100644
--- a/tests/snapshot/bytecode/class/methods.exp
+++ b/tests/snapshot/bytecode/class/methods.exp
@@ -42,7 +42,7 @@
     0: [BytecodeFunction: named1]
     1: [BytecodeFunction: named2]
     2: [BytecodeFunction: <anonymous>]
-    3: [BigInt: 1]
+    3: [BigIntValue: 1]
     4: [BytecodeFunction: <anonymous>]
     5: [BytecodeFunction: <anonymous>]
     6: [BytecodeFunction: get getter]
diff --git a/tests/snapshot/bytecode/expression/literal.exp b/tests/snapshot/bytecode/expression/literal.exp
index 88c1934..cd92ba8 100644
--- a/tests/snapshot/bytecode/expression/literal.exp
+++ b/tests/snapshot/bytecode/expression/literal.exp
@@ -13,6 +13,6 @@
   Constant Table:
     0: [String: ]
     1: [String: hello]
-    2: [BigInt: 1]
-    3: [BigInt: 1]
+    2: [BigIntValue: 1]
+    3: [BigIntValue: 1]
 }
