diff --git a/libcaf_core/caf/serialization.test.cpp b/libcaf_core/caf/serialization.test.cpp
index 6aeea610b..0e09829be 100644
--- a/libcaf_core/caf/serialization.test.cpp
+++ b/libcaf_core/caf/serialization.test.cpp
@@ -882,49 +882,53 @@ OUTLINE("serializing a type that initializes members to a non-empty state") {
   )_";
 }
 
-// TODO: update tests to use apply instead of value overload
 SCENARIO("binary serializer and deserializer handle vectors of booleans") {
   GIVEN("a binary serializer") {
     auto sink = binary_serializer_wrapper{sys};
     WHEN("serializing a vector of booleans with 8 values") {
       auto val = std::vector<bool>{true,  false, true, true,
                                    false, false, true, false};
-      check(sink.sink.value(val));
-      THEN("deserializing the result produces the value again") {
+      check(sink.sink.apply(val));
+      THEN("the serialized data is compressed into two bytes") {
+        // Two bytes: one for the varbyte encoding of the size, one for the
+        // packed boolean values.
+        check_eq(sink.buffer.size(), 2u);
+      }
+      AND_THEN("deserializing the result produces the value again") {
         auto source = binary_deserializer{sink.buffer, &sink.codec};
         auto copy = std::vector<bool>{};
-        check(source.value(copy));
+        check(source.apply(copy));
         check_eq(copy, val);
       }
     }
     WHEN("serializing a vector of boolean with 9 values") {
       auto val = std::vector<bool>{true,  false, true,  true, false,
                                    false, true,  false, true};
-      check(sink.sink.value(val));
+      check(sink.sink.apply(val));
       THEN("deserializing the result produces the value again") {
         auto source = binary_deserializer{sink.buffer, &sink.codec};
         auto copy = std::vector<bool>{};
-        check(source.value(copy));
+        check(source.apply(copy));
         check_eq(copy, val);
       }
     }
     WHEN("serializing a vector of boolean with 1 value") {
       auto val = std::vector<bool>{true};
-      check(sink.sink.value(val));
+      check(sink.sink.apply(val));
       THEN("deserializing the result produces the value again") {
         auto source = binary_deserializer{sink.buffer, &sink.codec};
         auto copy = std::vector<bool>{};
-        check(source.value(copy));
+        check(source.apply(copy));
         check_eq(copy, val);
       }
     }
     WHEN("serializing an empty vector of boolean") {
       auto val = std::vector<bool>{};
-      check(sink.sink.value(val));
+      check(sink.sink.apply(val));
       THEN("deserializing the result produces the value again") {
         auto source = binary_deserializer{sink.buffer, &sink.codec};
         auto copy = std::vector<bool>{};
-        check(source.value(copy));
+        check(source.apply(copy));
         check_eq(copy, val);
       }
     }
diff --git a/libcaf_core/caf/serializer.test.cpp b/libcaf_core/caf/serializer.test.cpp
index 1e4c662c1..f37493ac1 100644
--- a/libcaf_core/caf/serializer.test.cpp
+++ b/libcaf_core/caf/serializer.test.cpp
@@ -8,7 +8,6 @@
 
 #include "caf/actor_system.hpp"
 #include "caf/actor_system_config.hpp"
-#include "caf/detail/default_actor_handle_codec.hpp"
 
 #include <vector>
 
@@ -21,7 +20,7 @@ class test_serializer : public serializer {
 public:
   using super = serializer;
 
-  test_serializer(actor_system& sys, bool state) : codec_(sys), state_(state) {
+  test_serializer(bool state) : state_(state) {
     // nop
   }
 
@@ -40,7 +39,7 @@ public:
   }
 
   caf::actor_handle_codec* actor_handle_codec() override {
-    return &codec_;
+    return nullptr;
   }
 
   bool has_human_readable_format() const noexcept override {
@@ -93,6 +92,8 @@ public:
     return state_;
   };
 
+  using serializer::value;
+
   bool value(std::byte) override {
     return state_;
   };
@@ -162,7 +163,6 @@ public:
   };
 
 private:
-  detail::default_actor_handle_codec codec_;
   bool state_ = false;
   error err_;
 };
@@ -170,23 +170,23 @@ private:
 } // namespace
 
 TEST("base serializer") {
-  auto cfg = actor_system_config{};
-  auto sys = actor_system{cfg};
   SECTION("failing serializer") {
-    auto serializer = test_serializer{sys, false};
-    check(!serializer.begin_associative_array(3));
-    check(!serializer.end_associative_array());
-    check(!serializer.begin_key_value_pair());
-    check(!serializer.end_key_value_pair());
-    check(!serializer.list(std::vector<bool>{true, false}));
+    auto uut = test_serializer{false};
+    check(!uut.begin_associative_array(3));
+    check(!uut.end_associative_array());
+    check(!uut.begin_key_value_pair());
+    check(!uut.end_key_value_pair());
+    check(!uut.value(std::vector<bool>{true, false}));
   }
   SECTION("successful serializer") {
-    auto serializer = test_serializer{sys, true};
-    check(serializer.begin_associative_array(4));
-    check(serializer.end_associative_array());
-    check(serializer.begin_key_value_pair());
-    check(serializer.end_key_value_pair());
-    check(!serializer.list(std::vector<bool>{true, false, true}));
-    check(serializer.list(std::vector<bool>{true, false}));
+    auto uut = test_serializer{true};
+    check(uut.begin_associative_array(4));
+    check(uut.end_associative_array());
+    check(uut.begin_key_value_pair());
+    check(uut.end_key_value_pair());
+    // odd number of elements fails
+    check(!uut.value(std::vector<bool>{true, false, true}));
+    // even number of elements succeeds
+    check(uut.value(std::vector<bool>{true, false}));
   }
 }
